src/python/pants/source/filespec_test.py

```diff
@@ -24,36 +20,23 @@ def assert_rule_match(
             else:
                 self.create_file(expected_match)
         snapshot = self.request_single_product(Snapshot, PathGlobs([glob]))
-        if negate:
-            subset = set(expected_matches).intersection(set(snapshot.files))
-            assert subset == set()
-        else:
-            assert sorted(expected_matches) == sorted(snapshot.files)
+        assert sorted(expected_matches) == sorted(snapshot.files)
 
     def test_matches_single_star_0(self) -> None:
         self.assert_rule_match("a/b/*/f.py", ("a/b/c/f.py", "a/b/q/f.py"))
 
-    def test_matches_single_star_0_neg(self) -> None:
-        self.assert_rule_match("a/b/*/f.py", ("a/b/c/d/f.py", "a/b/f.py"), negate=True)
```

Comment:
Was it necessary to delete these? They do actually provide some coverage, because otherwise there is nothing that confirms that a glob doesn't match "everything". Alternatively, would need a three argument form of `assert_rule_match` that took `glob, paths, expected_matches`.

Reply:
Makes sense. I added them back, but refactored to hopefully be more clear.